hermes-client/src/test/groovy/pl/allegro/tech/hermes/client/HermesClientTest.groovy

```diff
@@ -120,6 +123,42 @@ class HermesClientTest extends Specification {
         latch.count == 2
     }
 
+    def "should wait until all sent after shutdown"() {
+        given:
+        CountDownLatch latch = new CountDownLatch(5)
+        HermesClient client = hermesClient(getCountDownDelayedSender(latch, 408, 20)).withRetries(5).build()
+
+        when:
+        client.publish(TOPIC, CONTENT_TYPE, CONTENT)
+
+        and:
+        client.shutdown()
+        client.awaitShutdown(50, 5, TimeUnit.SECONDS)
+
+        then:
+        latch.await()
```

Comment:
This test is going to hang indefinitely if some messages are not sent, right? How about calling `await` with some reasonable timeout? If it returns false then the test fails and we know it's because the condition wasn't met.